File: src/main/java/com/kh013j/model/domain/OrderedDish.java
package com.kh013j.model.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Data
@EqualsAndHashCode
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "orderdish", schema = "rh")
public class OrderedDish {
    @Id
    @SequenceGenerator(name = "orderdish-sequence_generator", sequenceName = "orderdish_sequence")
    @GeneratedValue(generator = "orderdish-sequence_generator", strategy = GenerationType.IDENTITY)
    long id;

    @ManyToOne
    @JoinColumn(name = "dish_id")
    private Dish dish;

    @ManyToOne
    @JoinColumn(name = "status_id", nullable = false)
    private Status status;

    private int quantity;

    public long getId() {
        return id;
    }

    public Dish getDish() {
        return dish;
    }

    public void setDish(Dish dish) {
        this.dish = dish;
    }

    public void setId(long id) {
        this.id = id;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }
}

File: front-end/src/app/models/orderedDish.ts
import {Dish} from "./dish";
import {Status} from "./status";
import {Order} from "./order";

export class OrderedDish {

  constructor(dish: Dish, status: Status, quantity: number) {
    this.dish = dish;
    this.status = status;
    this.quantity = quantity;
  }

  id: number;
  dish: Dish

  status: Status;
  quantity: number;
  order: Order;

  public setOrder(order: Order){
    this.order = order;
  }

  public toString(): String{
    return "OrderedDish: id" + this.id + ", dish: " + this.dish.toString() + ", status: " + this.status
    + ", quantity: " + this.quantity + ", order: " + this.order;
  }
}
